Source/FieldSolver/SpectralSolver/SpectralSolver.cpp

```diff
@@ -50,19 +53,30 @@ SpectralSolver::SpectralSolver(
     // - Select the algorithm depending on the input parameters
     //   Initialize the corresponding coefficients over k space
 
+    ParmParse pp("psatd");
+    pp.query("do_time_averaging", fft_do_time_averaging);
+
     if (pml) {
         algorithm = std::unique_ptr<PMLPsatdAlgorithm>( new PMLPsatdAlgorithm(
             k_space, dm, norder_x, norder_y, norder_z, nodal, dt ) );
-    } else if ((v_galilean[0]==0) && (v_galilean[1]==0) && (v_galilean[2]==0)){
-         // v_galilean is 0: use standard PSATD algorithm
-         algorithm = std::unique_ptr<PsatdAlgorithm>( new PsatdAlgorithm(
-             k_space, dm, norder_x, norder_y, norder_z, nodal, dt, update_with_rho ) );
-      } else {
-          // Otherwise: use the Galilean algorithm
-          algorithm = std::unique_ptr<GalileanAlgorithm>( new GalileanAlgorithm(
-              k_space, dm, norder_x, norder_y, norder_z, nodal, v_galilean, dt ));
-       }
-
+    }
+    else {
+        if (fft_do_time_averaging){
+              algorithm = std::unique_ptr<AvgGalileanAlgorithm>( new AvgGalileanAlgorithm(
+                k_space, dm, norder_x, norder_y, norder_z, nodal, v_galilean, dt ) );
+            }
+        else {
+            if ((v_galilean[0]==0) && (v_galilean[1]==0) && (v_galilean[2]==0)){
+               // v_galilean is 0: use standard PSATD algorithm
+              algorithm = std::unique_ptr<PsatdAlgorithm>( new PsatdAlgorithm(
+                   k_space, dm, norder_x, norder_y, norder_z, nodal, dt, update_with_rho ) );
+            }
+            else {
+                algorithm = std::unique_ptr<GalileanAlgorithm>( new GalileanAlgorithm(
+                    k_space, dm, norder_x, norder_y, norder_z, nodal, v_galilean, dt ) );
+          }
+            }
+        }
```

Comment:
Could you fix the indentation?
```suggestion
    }
    else {
        if (fft_do_time_averaging){
            algorithm = std::unique_ptr<AvgGalileanAlgorithm>( new AvgGalileanAlgorithm(
                k_space, dm, norder_x, norder_y, norder_z, nodal, v_galilean, dt ) );
        }
        else {
            if ((v_galilean[0]==0) && (v_galilean[1]==0) && (v_galilean[2]==0)){
                // v_galilean is 0: use standard PSATD algorithm
                algorithm = std::unique_ptr<PsatdAlgorithm>( new PsatdAlgorithm(
                   k_space, dm, norder_x, norder_y, norder_z, nodal, dt, update_with_rho ) );
            }
            else {
                algorithm = std::unique_ptr<GalileanAlgorithm>( new GalileanAlgorithm(
                    k_space, dm, norder_x, norder_y, norder_z, nodal, v_galilean, dt ) );
            }
        }
    }
```

Reply:
Fixed in commit 7e101bfd0667c57fa4420.